src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -279,14 +271,8 @@ private bool ParameterExistsInAllRoutes(ActionModel actionModel, string paramete
 
         private ModelMetadata GetParameterMetadata(ParameterModel parameter)
         {
-            if (_modelMetadataProvider is ModelMetadataProvider modelMetadataProvider)
-            {
-                return modelMetadataProvider.GetMetadataForParameter(parameter.ParameterInfo);
-            }
-            else
-            {
-                return _modelMetadataProvider.GetMetadataForType(parameter.ParameterInfo.ParameterType);
-            }
+            // No need for information from attributes on the parameter. Just use its type.
+            return _modelMetadataProvider.GetMetadataForType(parameter.ParameterInfo.ParameterType);
```

Comment:
Another thought that I had, is it worth changing up the design a bit to limit the leakage in this area. It would take the following steps.
1. Add an extension method on `GetMetadataForParameter(...)`IModelMetadataProvider 
2. Make the default metadata provider honour the new option
3. Use the extension method everywhere

That would put the onus on anyone implementing a metadata provider to honour the new options if they cared, but I don't think anyone implements that.

If you don't like this idea feel free to ignore. I think what you have here is pretty good already

Reply:
I'm not sold partly because this approach pushes the "fall back to `ModelMetadataKind.Type` metadata" behaviour down. And, it separates use of the option (in `DefaultModelMetadataProvider`) from `is ModelMetadataProvider` (in `ModelMetadataProviderExtensions`). But, mainly, I'd rather not try this out and rework the tests (where I'm expecting time-consuming fallout) since you're comfortable with the current approach.

We can certainly reconsider this later if (unexpectedly) we find other places needing these conditions. For one thing, `DefaultModelMetadataProvider` already has an `IOptions<MvcOptions>` constructor parameter. (Hmm, why isn't the other constructor `[Obsolete]`?)